src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1216,15 +1216,11 @@ format layout_optimizer::get_preferred_format(program_node& node) {
                 node.set_preferred_input_fmt(i, fmt);
             } else if (in_lay_rank != out_lay_rank) {
                 auto fmt = get_preferred_format(node.get_dependency(i));
-                // Check if selected format can be adjusted to the required input rank
+                // Check if selected format can be adjusted to the required input and output rank
```

Comment:
could you describe some more context here as a comment so that others can understand this logic faster?